.github/workflows/_e2e_tests.yaml

```diff
@@ -20,38 +31,73 @@ jobs:
       test_names: ${{ steps.set_test_names.outputs.test_names }}
     steps:
       - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
       - id: test_files
         name: Get test file names
         working-directory: test/e2e/
         # go list used to extract the test names from only those test files that
         # match the specified tags: here e2e_tests.
         # This filters out e.g. istio tests which we run separately.
         run: echo "result=$(go list -tags e2e_tests -f '{{ range .TestGoFiles }} {{- . }} {{ end }}' .)" >> $GITHUB_OUTPUT
+
       - name: Print test file names
         run: echo "Test file names ${{ steps.test_files.outputs.result }}"
+
       - id: set_test_names
         name: Set test names
         working-directory: test/e2e/
         # grep magic described in https://unix.stackexchange.com/a/13472
         # sed to add the extra $ is because some of our test names overlap. we need it so the -run regex only matches one test
         run: |
           echo "test_names=$(grep -shoP "(?<=^func )(Test[a-zA-z_0-9]+)(?=\(t \*testing.T\) {)" ${{ steps.test_files.outputs.result }} | sed -e "s/$/\$/"| jq -R . | jq -cs .)" >> $GITHUB_OUTPUT
+
       - name: Print test names
         run: echo "Test names ${{ steps.set_test_names.outputs.test_names }}"
-  e2e-kind-tests:
+
+  # TODO: extract this as a separate workflow reading the dependencies versions from a file (not only Kind, but also
+  # GKE, Istio, Kong, etc.)
+  # https://github.com/Kong/kubernetes-ingress-controller/pull/3724
```

Comment:
Can we create an issue for this instead linking to this PR?

Reply:
I made it an AC of https://github.com/Kong/kubernetes-ingress-controller/issues/3349.